modules/createHistory.js

```diff
@@ -117,11 +117,10 @@ function createHistory(options={}) {
       if (ok) {
         // treat PUSH to current path like REPLACE to be consistent with browsers
         if (nextLocation.action === PUSH) {
-          let { pathname, search } = getCurrentLocation()
-          let currentPath = pathname + search
-          let path = nextLocation.pathname + nextLocation.search
+          const prevPath = createPath(location)
```

Comment:
This is not 100% correct, since it doesn't e.g. account for `basename` when using browser history. In practice that's not an issue, though, since `basename` doesn't change.

This should be the "final" `createPath`, but I can't access that here.